pkg/controller/migration/namespace_migration.go

```diff
@@ -82,23 +83,26 @@ func (m *CoreNamespaceMigration) NeedsCoreNamespaceMigration() (bool, error) {
 		return false, nil
 	}
 
-	_, err := m.client.AppsV1().DaemonSets(kubeSystem).Get(nodeDaemonSetName, metav1.GetOptions{})
+	ctx := context.Background()
```

Comment:
I think you need to pass this context in as a function var to NeedsCoreNamespaceMigration